instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -251,13 +263,22 @@ def _before_cur_exec(
                         if self.commenter_options.get(k, True)
                     }
 
-                    statement = _add_sql_comment(statement, **commenter_data)
+                    if self.enable_attribute_commenter:
+                        statement = _add_sql_comment(
+                            statement, **commenter_data
+                        )
 
                 span.set_attribute(SpanAttributes.DB_STATEMENT, statement)
                 span.set_attribute(SpanAttributes.DB_SYSTEM, self.vendor)
                 for key, value in attrs.items():
                     span.set_attribute(key, value)
 
+                if (
+                    self.enable_commenter
+                    and not self.enable_attribute_commenter
+                ):
+                    statement = _add_sql_comment(statement, **commenter_data)
```

Comment:
Why do we need the _add_sql_comment after the set_attribute statement?

Reply:
@shalevr ! At this condition, user has enabled sqlcommenting but has not opted into "enable attribute commenter". So `db.statement` get written first without sqlcomment, then sqlcomment is added to the query statement before execution.